src/reducers/books.js

```diff
@@ -0,0 +1,22 @@
+// eslint-disable-next-line import/no-cycle
+import initState from '../index';
+
+const booksReducer = (state = initState || [], action) => {
+  if (action.type === 'CREATE_BOOK') {
+    const allBooks = [...state.books, action.book];
+    return {
+      ...state,
+      books: allBooks,
+    };
+  }
+  if (action.type === 'REMOVE_BOOK') {
+    const allBooks = state.books.filter(book => book.id !== action.book.id);
+    return {
```

Comment:
You are currently returning a copy of the old state together with a copy of the state minus the deleted book, this cause the rendering of the old `booklist` and the new `filtered books` when the state is updated but in actual fact we should only the list of filtered books **only**. I don't think ...state here is necessary. 🤔